### PR TITLE
Use lf EOL for travis and make files

### LINKED ISSUE #2152: Fake Travis Build files are not Cygwin-compatible
I have to change line endings from windows style to unix style in the configuration files, in order to be able to run the fake travis build in Cygwin.
I believe it could be fixed easily : 

> One way to fix this is to create .gitattributes file in repo's root and add lines like the following:
> `*.txt text eol=lf`

Thoughts ?

### PR DESCRIPTION
**Pull request by gaelj**
Fixes #2152

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -11,3 +11,7 @@ Makefile text
 *.bat eol=crlf
 *.txt text
 *.sh text
```
Comment (ledvinap) [inline .gitattributes]: just use
`*.sh eol=lf` - this implies `text` and it is reasonable to convert all shell scripts, not just travis

File: .gitattributes [hunk 2]
```diff
@@ -11,3 +11,7 @@ Makefile text
...
+fake_travis_build.sh eol=lf
+.travis.sh eol=lf
+Makefile eol=lf
```
Comment (ledvinap) [inline .gitattributes]: IMO this is pattern matching Makefile in any directory, so both match (`/Makefile` is pattern matching top-level one only)

**Comment by hydra** [comment]
squash please

**Comment by gaelj** [comment]
@hydra : branch is now squash-rebased and force pushed.
Thanks !

**Comment by martinbudden** [comment]
@gaelj , minor comment - the `.gitattributes` does not have an newline character at the end of the file. It's generally good practice to ensure text files end with newline (not having the newline character can cause problems with some linux tools).

**Comment by gaelj** [comment]
Oh right - I'll correct that later on today.

File: .gitattributes [hunk 3]
```diff
@@ -10,4 +10,5 @@
 Makefile text
 *.bat eol=crlf
 *.txt text
-*.sh text
+*.sh eol=lf
+Makefile eol=lf
```
Comment (ledvinap) [inline .gitattributes]: Makefile is referenced twice (text + eol) - it is technically correct, but possibly confusing

**Comment by gaelj** [comment]
Corrected

**Pull request merged**